lib/vm/src/libcalls.rs

```diff
@@ -453,6 +469,10 @@ impl LibCall {
             Self::RaiseTrap => wasmer_raise_trap as usize,
             Self::TruncF32 => wasmer_f32_trunc as usize,
             Self::TruncF64 => wasmer_f64_trunc as usize,
+            Self::Memory32Grow => wasmer_memory32_grow as usize,
```

Comment:
This is a bit confusing.
Were this libcalls not available before? How cranelift was using them?

Reply:
They were not libcalls before! This PR doesn't change how singlepass or cranelift use them, which is to use the "vm builtins" mechanism where every instance has pointers-to these functions as fields in the instance. See translate_memory_grow in codege-cranelift's func_environ.rs:
```
        let (vmctx, func_addr) = self.translate_load_builtin_function_address(&mut pos, func_idx);
        let call_inst = pos
            .ins()
            .call_indirect(func_sig, func_addr, &[vmctx, val, memory_index]);
```
`translate_load_builtin_function_address` will load the function pointer out of the instance itself (using VMBuiltinFunctionIndex on VMOffsets` and then return that function pointer.

IMO the direction should be that we remove the VMBuiltinFunction system entirely, unless you know of a good reason why we want to have function for different instances at runtime and don't know this at compile time.